cri/v1alpha2/cri.go

```diff
@@ -723,11 +732,26 @@ func (c *CriManager) CreateContainer(ctx context.Context, r *runtime.CreateConta
 		},
 		NetworkingConfig: &apitypes.NetworkingConfig{},
 	}
-	err := c.updateCreateConfig(createConfig, config, sandboxConfig, podSandboxID)
+
+	res, err := c.SandboxStore.Get(podSandboxID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get metadata of %q from SandboxStore: %v", podSandboxID, err)
+	}
+	sandboxMeta := res.(*SandboxMeta)
+	sandboxMeta.NetNS = containerNetns(sandbox)
+
+	err = c.updateCreateConfig(createConfig, config, sandboxConfig, sandboxMeta)
 	if err != nil {
 		return nil, err
 	}
 
+	// call cri plugin to update create config
+	if c.CriPlugin != nil {
+		if err := c.CriPlugin.PreCreateContainer(createConfig, sandboxMeta); err != nil {
```

Comment:
Why don't you put this hook immediately before `c.ContainerMgr.Create`? The changes of plugin would be overwrote by other code.

Reply:
Good idae.